Tests/Find-ImageName.Tests.ps1

```diff
@@ -1,13 +1,101 @@
 Import-Module -Force $PSScriptRoot/../Docker.Build.psm1
 
+function Get-TempPath {
+    if ($IsWindows) {
+        [system.io.path]::GetTempPath()
+    } elseif ($IsLinux) {
+        '/tmp'
+    }
+}
+
+function New-RandomFolder {
+    param (
+        [int]$FolderLength = 8
+    )
+    do {
+        $randomString = ( -join ((0x30..0x39) + ( 0x41..0x5A) + ( 0x61..0x7A) | Get-Random -Count $FolderLength | % {[char]$_}) )
+        $randomPath = Join-Path $(Get-TempPath) $randomString
+    } while (Test-Path -Path $randomPath -PathType Container)
+    New-Item -Path $randomPath -ItemType Directory | Out-Null
+    return $randomPath
+}
+
+function Fake-GitRepository {
+    param (
+        [ValidateNotNullOrEmpty()]
+        [String]$Path
+    )
+
+    $dotGitPath = Join-Path $Path ".git"
+    if (Test-Path $dotGitPath -PathType Container) {
+        Remove-Item $dotGitPath -Recurse -Force | Out-Null
+    }
+    New-Item $dotGitPath -ItemType Directory
+
+    $configData = @"
+[remote "origin"]
+url = https://github.com/3shapeAS/dockerbuild-pwsh.git
+"@
+
+    $configData | Out-File -FilePath "$dotGitPath/config" -Encoding ascii
+}
+
 Describe 'Parse context from git repository' {
 
+    Context 'Validating support functions for - When git is installed' {
+
+        It 'can get TEMP path' {
+            $tempPath = Get-TempPath
+
+            if ($IsWindows) {
+                $tempPath.IndexOf($Env:TEMP) | Should -Be 0
```

Comment:
When reading this, I wasn't sure what it is testing. Can you think of a way to make it clearer what you are verifying?

Reply:
To test if `[system.io.path]::GetTempPath()` == `$Env:TEMP` value on Windows, and `/tmp` on Linux.
I want to test `Get-TempFolder` function so it is able to be tested on both Windows and on Linux.